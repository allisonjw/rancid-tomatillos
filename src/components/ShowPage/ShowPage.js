import React, { Component } from 'react';
import MovieRatings from '../../containers/MovieRatings/MovieRatings'
import '../../containers/App/App.scss';

const ShowPage = () => {
  return(
    <section className="showPage-container">
      <h1 className="showPage-container-title">Title</h1>
      <p className="showPage-container-release">08/22/20</p>
      <p className="showPage-container-overview">blah blah blah</p>
      <p className="showPage-container-averageRating">average rating: 4</p>
      <MovieRatings />
    </section>
  )
}

export default ShowPage;
