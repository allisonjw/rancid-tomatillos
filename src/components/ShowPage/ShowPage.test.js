import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { ShowPage, mapStateToProps } from './ShowPage';


describe('ShowPage', () => {
  let wrapper;

  beforeEach(() => {

    wrapper = shallow(
        <ShowPage
        movie={{
            "id": 1,
            "title": "Jumanji: The Next Level",
            "poster_path": "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
            "backdrop_path": "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
            "release_date": "2019-12-04",
            "overview": "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game.",
            "average_rating": 6.666666666666667
        }}
        user={{email: 'diane@turing.io', name: 'Diane', id: 7}}
        />
    )

  })

  it('should render with correct data', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {

    it('should return the user object ', () => {
      const mockState = {
        user: {email: 'diane@turing.io', name: 'Diane', id: 7},
        movies: [{}, {}, {}],
        ratings: [{}, {}, {}]
      };
      const expected = {
        user: {email: 'diane@turing.io', name: 'Diane', id: 7}
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    })
  })
})
