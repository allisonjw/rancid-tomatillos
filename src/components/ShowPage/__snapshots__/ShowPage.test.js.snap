// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPage should render with correct data 1`] = `
<section
  className="showPage__image-container"
>
  <img
    alt="movie cover backdrop"
    className="showPage__image"
    src="https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg"
  />
  <div
    className="showPage__image-info-container"
  >
    <h3
      className="showPage-container-title"
    >
      Jumanji: The Next Level
    </h3>
    <p
      className="showPage-container-release"
    >
      <span>
        Release Date:
      </span>
      2019-12-04
    </p>
    <p
      className="showPage-container-overview"
    >
      <span>
        Synopsis:
      </span>
      In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game.
    </p>
    <p
      className="showPage-container-averageRating"
    >
      <span>
        Average Rating:
      </span>
      6.666666666666667
    </p>
    <Connect(MovieRatings)
      movieId={1}
    />
  </div>
</section>
`;
